mailpoet/lib/Config/Changelog.php

```diff
@@ -98,20 +108,47 @@ public function shouldShowRevenueTrackingPermissionPage() {
       && $this->wp->currentUserCan('administrator');
   }
 
+  public function redirectToLandingPage() {
+    if (isset($_GET['activate-multi'])) return; // do not redirect when activated with bulk activation mode
+
+    if ($this->shouldShowLandingPage() && !$this->isLandingPage()) {
+      $this->urlHelper->redirectTo(
+        $this->wp->adminUrl('admin.php?page=mailpoet-landingpage')
+      );
+    }
+  }
+
+  public function maybeRedirectToLandingPage() {
+    if ($this->isWelcomeWizardPage()) return; // do not redirect when on welcome wizard page
+
+    if ($this->shouldShowLandingPage()) {
```

Comment:
Redundant usage of `shouldShowLandingPage`, one of them can be removed.

Reply:
This method has an early return [here](https://github.com/mailpoet/mailpoet/blob/cd18db8499c2cd2f236f0e01242ac25a2a859c85/mailpoet/lib/Config/Changelog.php#L124-L129)

The landing page is currently an experimental feature. We must ensure the plugin redirects to the welcome wizard pending when the landing page is removed from experimental.

We can refactor the method to remove the duplicate check for `$this->shouldShowLandingPage()` when that is done.